src/CameraZoomPlugin.cc

```diff
@@ -352,31 +393,91 @@ void CameraZoomPlugin::PreUpdate(
   if (!comp)
     return;
 
-  if (!this->impl->zoomChanged)
+  if (this->impl->zoomChanged) {
+    // Only calculate goal once each time zoom is changed.
+    const auto requestedZoomCmd = this->impl->zoomCommand.load();
+    const auto clampedZoomCmd = std::clamp(requestedZoomCmd,
+      this->impl->minZoom, this->impl->maxZoom);
+    if (std::abs(requestedZoomCmd - clampedZoomCmd) >
+      std::numeric_limits<double>::epsilon())
+    {
+      gzwarn << "Requested zoom command of " << requestedZoomCmd
+        << " has been clamped to " << clampedZoomCmd << ".\n";
+    }
+    this->impl->goalHfov =  this->impl->curHfov / clampedZoomCmd;
```

Comment:
extra space around `=`.

Reply:
Fixed all of them I could find through search in aebe9e42e7eb688590c1adde6a84be16b8cff06b